xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
BTW it makes things really complicated and therefor you should use the forearch syntax with the break inside the loop or revert the changes to the `EPGMAP::const_iteration` for syntax like it was before.